### PR TITLE
Uart add port 1 PINs support

### PR DESCRIPTION
**Pull request by jakub-uC**
1, UART driver is capable to configure UART lines either or port 0 or port 1.
2. GPIO handler changed to Nordic HAL.
3. Added missing baudrates supported by Nordic HAL

### CODE REVIEW

**Review by ioannisg — COMMENTED** [review]
Does GPIO driver support P1?
If so, I do not see clearly why we need to change the behavior of the UART driver and use nrf_gpio functions directly, instead of gpio_ generic functions.

**Comment by jarz-nordic** [comment]
Because it will help us keep code cleaner and easier to maintain.

Otherwise we would have to put some `ifdefs` in the init function. I tried it in UARTE driver but @anangl proposed nrf_gpio approach. When I compared both solutions, I like nrf_gpio support much better.

**Comment by anangl** [comment]
Changes in code are okay. But I'd rephrase the first commit message. One may wonder what is this "GPIO handler". And actually it is a single change. You are providing the proper handling of pins from GPIO port 1 right by replacing the GPIO driver API calls with the ones to Nordic GPIO HAL.

**Review by carlescufi — CHANGES_REQUESTED** [review]
We really need to explain properly, both in the code and in the commit message, why we are switching away from the standard GPIO functions and into our own. As it stands now there's no explanation anywhere

**Comment by ioannisg** [comment]
Agreeing with @carlescufi

**Comment by anangl** [comment]
The dependency on the GPIO driver was a bit artificial, since the UART driver used the GPIO API functions only to configure the pins at its initialization. This configuration can be done by using nrfx GPIO HAL functions directly (thus, just like in case when a shim is using an nrfx driver underneath). Additionally, the HAL functions can be fed with the absolute pin numbers (that are currently used for Nordic SoCs in Kconfig and DT) and for the GPIO driver an additional translation would be required, to extract the port and its relative pin number.
And finally, the UART driver has to be initialized at PRE_KERNEL_1 (because of the console using it) and currently the GPIO driver is initialized at POST_KERNEL stage, so calling its API function form the UART initialization routine was incorrect.

**Comment by carlescufi** [comment]
@anangl thanks for the info. @jarz-nordic can you please update with this info/explanation both as a code comment and also in the commit?

**Comment by anangl** [comment]
@carlescufi I am not convinced that putting this info into code is necessary. There would be no trace of the GPIO driver in the UART driver code, so why to refer to something that is no longer present? And the code using just GPIO HAL functions will be easy to understand. I see no need for extra explanation of it.
IMO the info in the commit that introduces the change should be sufficient.

**Comment by anangl** [comment]
What about updating the commit title as well?
To something like "drivers: serial: nrf: Use nrfx GPIO HAL to properly handle pins from P1" maybe?

**Comment by carlescufi** [comment]
@jarz-nordic yes please update the commit title

**Comment by anangl** [comment]
"API function form" -> "API functions from" in the last paragraph of the first commit message.
Apart from that, LGTM.

**Review by anangl — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
